corehq/apps/reports/standard/sms.py

```diff
@@ -388,11 +388,20 @@ def order_by_col(data_):
         queryset = SMS.objects.filter(
             domain=self.domain,
             date__range=(self.datespan.startdate_utc, self.datespan.enddate_utc),
-        ).exclude(
-            # Exclude outgoing messages that have not yet been processed
-            direction=OUTGOING,
-            processed=False
         )
+        if toggles.INCLUDE_SMS_ERRORS.enabled(self.request.user.username):
+            queryset = queryset.exclude(
+                direction=OUTGOING,
+                processed=False,
+                error=False,  # Don't exclude errored messages
```

Comment:
Is this because errored messages are never processed?

Reply:
I believe so - the project in question has 10 out of 227 total messages being excluded by the clause here, and adding `error=False` causes them all to appear in results.  It may be a simple oversight that those errored messages aren't being marked as `processed`, as I didn't get the impression that any processing is still pending (the failures ranged from opt-outs to gateway issues).  That might also be the intended behavior - like "processed" means "processed successfully" or something.  I'm not familiar enough with this code to say for sure.

Reply:
although from the comment
> Exclude outgoing messages that have not yet been processed 

I'm fairly certain the original intent was not to exclude error messages.  It's been like that for 4+ years though, so I'm playing it safe to see what the ramifications are.